src/live_restore/live_restore_fs.c

```diff
@@ -580,22 +589,36 @@ __live_restore_remove_extlist_hole(
     return (0);
 }
 
-/*
+typedef enum { FULL, PARTIAL, NONE } WT_LIVE_RESTORE_SERVICE_STATE;
+
+/* !!!
  * __live_restore_can_service_read --
- *     Return if a read can be serviced by the destination file. This assumes that the block manager
- *     is the only thing that perform reads and it only reads and writes full blocks. If that
- *     changes this code will unceremoniously fall over.
+ *     Return if a read can be serviced by the destination file. Callers must hold the extent list
+ *     read lock at a minimum.
+ *     There are three possible scenarios:
+ *     - The read is entirely within a hole and we return NONE.
+ *     - The read is entirely outside of all holes and we return FULL.
+ *     - The begins outside a hole and then ends inside, in which case we return PARTIAL.
+ *       This will only happen if a background migration thread is copying data and has partially
+ *       the migrated content we're reading. The background threads always copy data in order,
```

Comment:
Two minor issues:
1. Please define the 'This' at the start of the sentence to "This situation" or "This case" or "This scenario"...
2. Something is missing in "...has partially the migrated content..."

Reply:
12390ade1